Detectors/EMCAL/reconstruction/include/EMCALReconstruction/CaloRawFitter.h

```diff
@@ -0,0 +1,156 @@
+  int getNsamplePed() const { return mNsamplePed; }
+
+  // access to array info
+  double getReversed(const int i) const { return mReversed[i]; }
+  const char* getAlgoName() const { return mName.c_str(); }
+  const char* getAlgoAbbr() const { return mNameShort.c_str(); }
+  FitAlgorithm getAlgo() const { return mAlgo; }
+
+  /// \brief Get the maximum of a bunch array
+  /// \return Maximum amplitute
+  short maxAmp(const AltroDecoder::Bunch* bunch, int* maxindex) const;
+
+  /// \brief Get maximum of array
+  /// \return Maximum amplitute
+  unsigned short maxAmp(const unsigned short* data, const int length) const;
+
+  /// \brief A bunch is considered invalid if the maximum is in the first or last time-bin.
+  bool checkBunchEdgesForMax(const AltroDecoder::Bunch* const bunch) const;
+
+  /// \brief Check if the index of the max ADC vaue is consistent with trigger.
+  bool isInTimeRange(const int maxindex, const int maxtime, const int mintime) const;
+
+  /// \brief Time sample comes in reversed order, revers them back Subtract the baseline based on content of altrocfg1 and altrocfg2.
+  /// \return Pedestal
+  float reverseAndSubtractPed(const AltroDecoder::Bunch* bunch,
+                              std::optional<unsigned int> altrocfg1, std::optional<unsigned int> altrocfg2,
+                              double* outarray) const;
+  /// \brief We select the bunch with the highest amplitude unless any time constraints is set.
+  /// \return The index of the array with the maximum aplitude
+  int selectBunch(const std::vector<AltroDecoder::Bunch>& bunchvector,
+                  short* maxampbin, short* maxamplitude);
+
+  /// \brief Selection of subset of data from one bunch that will be used for fitting or Peak finding.
+  /// Go to the left and right of index of the maximum time bin
+  /// Until the ADC value is less than cut, or derivative changes sign (data jump)
+  void selectSubarray(const double* data, int length, short maxindex,
+                      int* first, int* last, int cut) const;
+
+  /// \brief Pedestal evaluation if not zero suppressed
+  float evaluatePedestal(std::vector<uint16_t> data, std::optional<int> length) const;
+
+  /// \brief Calculates the chi2 of the fit
+  ///
+  /// \param amp    - max amplitude;
+  /// \param time   - time of max amplitude;
+  /// \param first  - sample array indices to be used
+  /// \param last   - sample array indices to be used
+  /// \param adcErr - nominal error of amplitude measurement (one value for all channels) if adcErr<0 that mean adcErr=1.
+  /// \param tau    - filter time response (in timebin units)
+  ///
+  /// \return chi2
+  double calculateChi2(double amp, double time,
+                       int first, int last,
+                       double adcErr = 1,
+                       double tau = 2.35) const;
+
+ protected:
+  double mReversed[constants::EMCAL_MAXTIMEBINS]; ///< Reversed sequence of samples (pedestalsubtracted)
```

Comment:
std::array, should only be class member if information is shared within several methods.